q2cli/tools.py

```diff
@@ -37,6 +37,21 @@ def export_data(path, output_dir):
     result.export_data(output_dir)
 
 
+def show_importable_types(ctx, param, value):
+    import qiime2.sdk
+
+    PluginManager = qiime2.sdk.PluginManager()
+    set_of_importable_types = PluginManager.importable_types
+    set_of_importable_types = [repr(t) for t in set_of_importable_types]
+    set_of_importable_types = sorted(set_of_importable_types)
+
+    if not value or ctx.resilient_parsing:
```

Comment:
Can you move this `if` statement to the very top of the method? (should come before `import qiime2.sdk` because that can be slow). That way we let Click short-circuit the method without doing any extra work.